### PR TITLE
Mini book Docker

### PR DESCRIPTION
**Pull request by Dantemss**
- Made mini ecosystem a factory
- Use the factory in tests
- Added a demo[mini] config
- demo[mini] can run without Exercises being configured
- Added a new Dockerfile, docker-compose.yml and README

Known issues:

- The CSP probably needs to be updated.

### CODE REVIEW

**Comment by Dantemss** [comment]
Migration test is failing because it's trying to run `demo[mini]` without the exercises key/secret and that is not allowed in the old code (and it did not even exist back then). Once this PR is merged, it should start passing.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+ARG NODE_JS_MAJOR_VERSION
+
+# The nodejs setup script already calls apt-get update,
+# but just in case they stop doing it we call it here again
+RUN wget --quiet --output-document=- https://deb.nodesource.com/setup_$NODE_JS_MAJOR_VERSION.x | \
+    bash - && \
+    apt-get update && \
+    apt-get install --yes libjemalloc2 nodejs google-chrome-stable && \
+    rm --recursive --force /var/lib/apt/lists/*
```
Comment (nathanstitt) [inline docker/Dockerfile:18]: this is failing with invalid signatures for me:
```
Step 6/17 : RUN wget --quiet --output-document=- https://deb.nodesource.com/setup_$NODE_JS_MAJOR_VERSION.x |     bash - &&     apt-get update &&     apt-get install --yes libjemalloc2 nodejs google-chrome-stable &&     rm --recursive --force /var/lib/apt/lists/*
 ---> Running in b32619da2121

## Installing the NodeSource Node.js 14.x repo...


## Populating apt-get cache...

+ apt-get update
Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:2 http://deb.debian.org/debian buster InRelease [121 kB]
Get:3 https://dl.google.com/linux/chrome/deb stable InRelease [1811 B]
Err:3 https://dl.google.com/linux/chrome/deb stable InRelease
  At least one invalid signature was encountered.
Get:4 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
Err:1 http://security.debian.org/debian-security buster/updates InRelease
  At least one invalid signature was encountered.
Err:2 http://deb.debian.org/debian buster InRelease
  At least one invalid signature was encountered.
Err:4 http://deb.debian.org/debian buster-updates InRelease
  At least one invalid signature was encountered.
```

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+[Tutor Docker Quickstart](../docker-quickstart.md)
```
Comment (nathanstitt) [inline docker/README.md:1]: we could probably remove this file

File: docker/Dockerfile (hunk 1 above)
Comment (nathanstitt) [reply docker/Dockerfile:18]: turned out I must have had some stale layers in my docker.  I ran `docker image prune` and fixed it (and also freed 32GB)

**Comment by nathanstitt** [comment]
@Dantemss I've got it running and seems to work well, but it didn't add any users.  I was able to add one from the dev login but there's no way to make him a verified instructor.  Can we add at least teacher01 and student01 and give them some tasks if it's easy?

**Review by nathanstitt — APPROVED** [review]
I've been testing and works really well.  Thanks @Dantemss !

**Comment by Dantemss** [comment]
Might need to increase the timeout on the action now that we have 2 mini books (seems like 1 minute is not enough anymore)

**Pull request merged**